node/narwhal/tests/common/primary.rs

```diff
@@ -209,6 +209,33 @@ impl TestNetwork {
         sleep(Duration::from_millis(MAX_BATCH_DELAY * 2)).await;
         self.validators.values().all(|v| v.primary.current_round() <= halt_round)
     }
+
+    // Checks if the committee is coherent in storage for all nodes (not quorum).
+    pub fn is_committee_coherent<T>(&self, rounds_range: T) -> bool
```

Comment:
I made these functions since they might end up being useful for other e2e tests. Note they also verify the conditions for all nodes in the network, not just the subset making up quorum. It may be better to implement the latter in future if/when we write cases that include byzantine nodes. For now, I'm assuming all the nodes are honest.

Reply:
Alternatively, we could go the other way and check that all nodes have reached the round before asserting on state (with a new helper similar to `is_round_reached`, perahps `is_round_reached_by_all`).